File: Лаба2/Задание 3 вариант 3/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	input, err := reader.ReadString('\n')
	if err != nil {
		fmt.Println("Failed to read input:", err)
		os.Exit(1)
	}

	inputs := strings.Fields(input)

	n, err := strconv.Atoi(inputs[0])
	if err != nil {
		fmt.Println("Invalid input:", err)
		os.Exit(1)
	}

	count := 0
	for i := 1; i <= n; i++ {
		g, err := strconv.Atoi(inputs[i])
		if err != nil {
			fmt.Println("Invalid input:", err)
			os.Exit(1)
		}
		number := g
		for number > 0 {
			h := number % 10
			number /= 10
			if h%3 == 0 {
				count += h
			}
		}
	}

	fmt.Println(count)
}

File: Лаба2/Задание 3 вариант 3/main.rs
use std::io;

fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Failed to read line");
    let inputs: Vec<&str> = input.trim().split_whitespace().collect();
    let n: usize = inputs[0].parse().expect("Invalid input");

    let mut count = 0;
    for i in 1..=n {
        let g: i32 = inputs[i].parse().expect("Invalid input");
        let mut number = g;
        while number > 0 {
            let h = number % 10;
            number /= 10;
            if h % 3 == 0 {
                count += h;
            }
        }
    }

    println!("{}", count);
}
